### PR TITLE
Update drift velocity calculation to include an additional optional model

### PR DESCRIPTION
**Pull request by brucehoward-physics**

### CODE REVIEW

**Comment by lgarren** [comment]
code-checks

File: lardataalg/DetectorInfo/DetectorPropertiesStandard.cxx [hunk 1]
```diff
@@ -216,54 +216,71 @@ namespace detinfo {
...
+    if (fUseIcarusMicrobooneDriftModel) {
+      constexpr double P0 = 0.;
+      constexpr double P1 = 5.53416;
+      constexpr double P2 = -6.53093;
+      constexpr double P3 = 3.20752;
+      constexpr double P4 = 0.389696;
+      constexpr double P5 = -0.556184;
+      vd = (1.0 - 0.0184 * (temperature - 89.0)) *
+           (P0 + P1 * std::pow(efield, 1) + P2 * std::pow(efield, 2) + P3 * std::pow(efield, 3) +
+            P4 * std::pow(efield, 4) + P5 * std::pow(efield, 5));
```
Comment (knoepfel) [inline lardataalg/DetectorInfo/DetectorPropertiesStandard.cxx]: Recommend replacing `std::pow(efield, n)` run-time calculations with more efficient template-based ones:

```c++
#include "cetlib/pow.h"

vd = ...
     (P0 + P1 * cet::pow<1>(efield) + P2 * cet::pow<2>(efield) + P3 * cet::pow<3>(efield) +
      P4 * cet::pow<4>(efield) + P5 * cet::pow<5>(efield));
```

**Review by knoepfel — COMMENTED** [review]
Thanks, @brucehoward-physics, for the PR.  Only one comment really--there is a more efficient template-based power function in `cetlib` for integral values.  See specifics below.

**Comment by brucehoward-physics** [comment]
Thanks for the comment Kyle -- that's good to know!

**Review by knoepfel — COMMENTED** [review]
Looks good, Bruce.  Thanks.  Will approve after L2 approval.

**Comment by lgarren** [comment]
code-checks

**Comment by lgarren** [comment]
trigger build

**Comment by lgarren** [comment]
trigger build

**Comment by lgarren** [comment]
There is a problem in the dunetpc unit tests.

**Comment by lgarren** [comment]
UseIcarusMicrobooneDriftModel is supposed to be optional, but dunetpc test_ExponentialChannelNoiseService fails with this:
`  Missing parameters:
   -    UseIcarusMicrobooneDriftModel: <bool>
`

**Comment by brucehoward-physics** [comment]
Is it using a fcl that doesn't include `detectorproperties.fcl` from lardataalg ? I added the parameter to default as false in this fcl: https://github.com/brucehoward-physics/lardataalg/blob/feature/howard_driftVel/lardataalg/DetectorInfo/detectorproperties.fcl#L20

File: lardataalg/DetectorInfo/DetectorPropertiesStandard.h [hunk 2]
```diff
@@ -108,6 +108,11 @@ namespace detinfo {
         Comment("Allows a scaling factor to fudge the drift velocity "
                 "calculation (as suggested by DriftVel Stancari")};
 
+      fhicl::Atom<bool> UseIcarusMicrobooneDriftModel{
+        Name("UseIcarusMicrobooneDriftModel"),
+        Comment("Allows user to decide to use the ICARUS+MicroBooNE drift "
+                "model for velocity calculation as in arXiv:2008.09765")};
```
Comment (knoepfel) [inline lardataalg/DetectorInfo/DetectorPropertiesStandard.h]: Suggest adding the default value `false` at the parameter declaration here so people who do not use the `detectorproperties.fcl` file below can continue to use their code as is--i.e. preserve backwards compatibility.

**Comment by brucehoward-physics** [comment]
Sure - I wasn't sure how folks felt about having a code-defined default vs. fcl-file-defined.

To confirm, it is simply adding a bare `false` i.e.
```
fhicl::Atom<bool> UseIcarusMicrobooneDriftModel{
        Name("UseIcarusMicrobooneDriftModel"),
        Comment("Allows user to decide to use the ICARUS+MicroBooNE drift "
                "model for velocity calculation as in arXiv:2008.09765"),
        false};
```
?

**Comment by knoepfel** [comment]
That's correct @brucehoward-physics.

**Comment by lgarren** [comment]
trigger build

testing mechanism

**Comment by brucehoward-physics** [comment]
Thanks Kyle, I'll whip up a quick commit with that change.

**Comment by lgarren** [comment]
trigger build

**Comment by lgarren** [comment]
The CI tests appear to be normal

**Comment by absolution1** [comment]
approve

**Comment by lgarren** [comment]
approve

**Pull request merged**